widgetssdk/src/main/java/com/glia/widgets/core/dialog/DialogController.kt

```diff
@@ -17,9 +15,6 @@ internal interface DialogContract {
         fun dismissDialogs()
         fun showExitQueueDialog()
         fun showExitChatDialog()
-        fun showUpgradeAudioDialog(mediaUpgradeOffer: MediaUpgradeOffer, operatorName: String?)
```

Comment:
Why have you divided the dialog logic? To my mind the dialog controller and state should remain more or less unchanged as the Dialog UI creator and restoration classes.

Reply:
I hesitated on this question for a long time. DialogController aims to handle dialog queues, priorities, and cross-screen dialogs, but we're dealing with none of this here. The same was in `EngagementCompletionActivityWatcher`. So my feeling is that `DialogController` adds complexity in these two cases.
I would like more opinions on this matter.

Reply:
I think I will return these to DialogController